src/main/java/frc/robot/commands/IntakeToAmperCommand.java

```diff
@@ -11,16 +39,82 @@ public class IntakeToAmperCommand extends Command {
+      .withDeadband(MAX_VELOCITY.times(DEADBAND).in(MetersPerSecond))
+      .withRotationalDeadband(MAX_ANGULAR_VELOCITY.times(DEADBAND).in(RadiansPerSecond))
+      .withDriveRequestType(DriveRequestType.Velocity)
+      .withSteerRequestType(SteerRequestType.MotionMagic);
+
+  // Desired chassis speeds. Defined here to prevent reallocation.
+  private ChassisSpeeds desiredChassisSpeeds = new ChassisSpeeds();
 
-  public IntakeToAmperCommand(IntakeSubsystem intakeSubsystem, AmperSubsystem amperSubsystem) {
+
+  public IntakeToAmperCommand(
+  IntakeSubsystem intakeSubsystem,  
+  AmperSubsystem amperSubsystem,
+  CommandSwerveDrivetrain drivetrainSubsystem,
+  LimelightSubsystem limelightSubsystem, 
+  Supplier<Measure<Velocity<Distance>>> translationXSupplier,
+  Supplier<Measure<Velocity<Distance>>> translationYSupplier,
+  Supplier<Measure<Velocity<Angle>>> rotationSupplier,
+  LimelightResults limelightResults,
+  LimelightCalcs limelightCalcs,
+  LimelightDetectorTarget limelightDetectorTarget) {
     this.intakeSubsystem = intakeSubsystem;
     this.amperSubsystem = amperSubsystem;
+    this.drivetrainSubsystem = drivetrainSubsystem;
+    this.translationXSupplier = translationXSupplier;
+    this.translationYSupplier = translationYSupplier;
+    this.rotationSupplier = rotationSupplier;
+    this.limelightResults = limelightResults;
+    this.limelightCalcs = limelightCalcs;
+    this.limelightDetectorTarget = limelightDetectorTarget;
+
+    addRequirements(intakeSubsystem, amperSubsystem, drivetrainSubsystem);
+  }
+
+public Rotation2d getTargetRotation() {
+  return limelightCalcs.getRobotRelativeTargetInfo(
+  limelightDetectorTarget.targetXDegrees, limelightDetectorTarget.targetYDegrees).angle;
+}
 
-    addRequirements(intakeSubsystem, amperSubsystem);
+  @Override 
+  public void initialize() {
+    // Reset the slew rate limiters, in case the robot is already moving
+    rateLimiter.reset(drivetrainSubsystem.getCurrentFieldChassisSpeeds());
   }
 
   @Override
   public void execute() {
+    desiredChassisSpeeds.vxMetersPerSecond = translationXSupplier.get().in(MetersPerSecond);
+    desiredChassisSpeeds.vyMetersPerSecond = translationYSupplier.get().in(MetersPerSecond);
+    desiredChassisSpeeds.omegaRadiansPerSecond = rotationSupplier.get().in(RadiansPerSecond);
+    var limitedCassisSpeeds = rateLimiter.calculate(desiredChassisSpeeds);
+    drivetrainSubsystem.setControl(drive
+        .withVelocityX(limitedCassisSpeeds.vxMetersPerSecond)
+        .withVelocityY(limitedCassisSpeeds.vyMetersPerSecond)
+        .withRotationalRate(limitedCassisSpeeds.omegaRadiansPerSecond));
+    if (limelightResults.valid == true) {
+      desiredChassisSpeeds.vxMetersPerSecond = translationXSupplier.get().in(MetersPerSecond);
+      desiredChassisSpeeds.vyMetersPerSecond = translationYSupplier.get().in(MetersPerSecond);
+      desiredChassisSpeeds.omegaRadiansPerSecond = getTargetRotation().getRadians();
```

Comment:
I think we need to use a PIDController (or preferably, a ProfiledPIDController) here. We don't just want to rotate at a speed equal to the target's angle.